### PR TITLE
Bg user story 11

### LINKED ISSUE #14: User Story 11, User Registration Missing Details
As a visitor
When I visit the user registration page
And I do not fill in this form completely,
I am returned to the registration page
And I see a flash message indicating that I am missing required fields

### PR DESCRIPTION
**Pull request by brian-greeson**
- Updated UserController with new and create actions
- Updated User and UserDetail models to validate the association and accept nested params
- Remove path expectations for new view from test.

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-  <%= text_field_tag :name %>
+<%= form_tag '/profile', method: :post do |f| %>
+  <%= label_tag :name, "Name" %>
+  <%= text_field_tag "user_detail_attributes[name]", "#{@user.user_detail.name}" %>
   <%= label_tag :street_address, "Street Address" %>
-  <%= text_field_tag :street_address %>
+  <%= text_field_tag "user_detail_attributes[street_address]", "#{@user.user_detail.street_address}" %>
   <%= label_tag :city %>
-  <%= text_field_tag :city %>
+  <%= text_field_tag "user_detail_attributes[city]", "#{@user.user_detail.city}" %>
```
Comment (kelshadarby) [inline app/views/users/new.html.erb:7]: Nice! This is a great way to go about this

File: app/models/user.rb [hunk 2]
```diff
@@ -2,11 +2,13 @@ class User < ApplicationRecord
   attr_accessor :password_confirmation
 
   validates_presence_of :email_address
-  validates :email_address, uniqueness: true
+  validates :email_address, uniqueness: {message: "Email is already in use."}
```
Comment (kelshadarby) [inline app/models/user.rb:5]: This is a great place for this

File: spec/features/users/new_spec.rb [hunk 3]
```diff
@@ -1,24 +1,107 @@
...
+      click_button "Submit"
+      
+      expect(current_path).to eq '/profile'
+      expect(page).to have_content("Registration sucessful! You are now logged in.")
+    end
+    it "cannot register a user if form is incomplete" do
+      visit '/register'
+
+      fill_in "user_detail_attributes[name]", with: @user1.user_detail.name
+      # fill_in "user_detail_attributes[street_address]", with: @user1.user_detail.street_address
```
Comment (kelshadarby) [inline spec/features/users/new_spec.rb:53]: Very easy to read

**Review by fredrondina — COMMENTED** [review]
Great job cleaning up the user create action. Looks much cleaner now than it was. 
Nice solution for pre-populating the new user form, it works great.

**Comment by brian-greeson** [comment]
Closes: #14

**Pull request merged**